src/xmerl_dsig.erl

```diff
@@ -0,0 +1,730 @@
+            logger:debug("SigInfoCanonData:~n~p~n", [Data]),
+
+            [#xmlText{value = Sig64}] =
+                xmerl_xpath:string(
+                  "//*[\"SignatureValue\"=local-name() and \"http://www.w3.org/2000/"
+                  "09/xmldsig#\"=namespace-uri()]/text()", Signature),
+            Sig = base64:decode(Sig64),
+
+            [#xmlText{value = Cert64}] =
+                xmerl_xpath:string(
+                  "//*[\"X509Certificate\"=local-name() and \"http://www.w3.org/2000/"
+                  "09/xmldsig#\"=namespace-uri()]/text()", Signature),
+            CertBin = base64:decode(Cert64),
+            CertHash = crypto:hash(sha, CertBin),
+            CertHash2 = crypto:hash(sha256, CertBin),
+
+            Cert = public_key:pkix_decode_cert(CertBin, plain),
+            #'Certificate'{
+               tbsCertificate =
+                   #'TBSCertificate'{
+                      subjectPublicKeyInfo =
+                          #'SubjectPublicKeyInfo'{
+                             subjectPublicKey =
+                                 PubKey}}} = Cert,
+            KeyBin = case PubKey of
+                         {_, KeyBin2} -> KeyBin2;
+                         KeyBin3 -> KeyBin3
+                     end,
+            Key = public_key:pem_entry_decode({'RSAPublicKey', KeyBin,
+                                               not_encrypted}),
+            case public_key:verify(Data, HashFunction, Sig, Key) of
+                true ->
+                    case Fingerprints of
+                        any ->
+                            verify_signatures(Tail, Element, Fingerprints);
+                        [#'OTPCertificate'{}|_] ->
+                            logger:debug("Cert: ~p~n",
+                                      [public_key:pkix_decode_cert(CertBin, otp)]),
+                            case lists:any(
+                                   fun(CA) ->
+                                           public_key:pkix_is_issuer(
+                                             CertBin, CA)
+                                   end, Fingerprints) of
+                                true ->
+                                    verify_signatures(Tail, Element, Fingerprints);
+                                false ->
+                                    {error, cert_not_accepted}
+                            end;
+                        _ ->
+                            case lists:any(
+                                   fun(X) ->
+                                           lists:member(X, Fingerprints)
+                                   end, [CertHash, {sha,CertHash},
+                                         {sha256,CertHash2}]) of
+                                true ->
+                                    verify_signatures(Tail, Element, Fingerprints);
+                                false ->
+                                    {error, cert_not_accepted}
+                            end
```

Comment:
Is this case common? Can we not just start out supporting a list of trusted issuer certificates before adding cases like this? Or at least provide three API calls: `verify_trusting_issuers`, `verify_trusting_fingerprints`, `verify_trusting_anyone`.